src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/AbstractRazorDelegatingEndpoint.cs

```diff
@@ -105,6 +116,8 @@ protected virtual Task<TResponse> HandleDelegatedResponseAsync(TResponse delegat
             return default;
         }
 
+        var correlationId = Guid.NewGuid();
+        using var __ = _telemetryReporter?.TrackLspRequest(nameof(HandleRequestAsync), LspName, LanguageServerConstants.RazorLanguageServerName, correlationId);
```

Comment:
For telemetry I recommend hardcoding instead of using `nameof`. That way, even though it's unlikely, if the method name changes we don't break telemetry